### PR TITLE
Use FunctionArgumentDescriptors to check bitSlice function + add test case

### PR DESCRIPTION
**Pull request by Blokje5**
### Changelog category (leave one):
Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Use FunctionArgumentDescriptors for bitSlice

Related Issue: #67447

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 8d979680060f10e6bcec3fc83fb3bdbaa7bb3deb with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67674/8d979680060f10e6bcec3fc83fb3bdbaa7bb3deb/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67674/8d979680060f10e6bcec3fc83fb3bdbaa7bb3deb/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/03214_bitslice_argument_evaluation.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- No arguments passed
```
Comment (alexkats) [inline tests/queries/0_stateless/03214_bitslice_argument_evaluation.sql:1]: Let's add `-- { echo }` in the beginning, so it'll be easier to debug this test in the future

**Review by alexkats — COMMENTED** [review]
looks good, thanks, one minor thing from my end

**Comment by alexkats** [comment]
Also, style check has failed, you need to remove unused error codes

**Comment by Blokje5** [comment]
Thanks for the review @alexkats, I have pushed an update that fixes the style check + adds the debug statement.

**Comment by Blokje5** [comment]
It seems one test failed, although it is unclear why. Any suggestions?

**Comment by Blokje5** [comment]
Looking at the issues, it might be related to: #67788

**Review by alexkats — APPROVED** [review]

**Comment by alexkats** [comment]
> Looking at the issues, it might be related to: #67788

No, this failure is different. There's actually a data race happened. I took a look, it's not related to the changes.

**Comment by alexkats** [comment]
Stateless tests (tsan, s3 storage) [3/4] - #67796

File: src/Functions/bitSlice.cpp [hunk 2]
```diff
@@ -40,28 +38,22 @@ class FunctionBitSlice : public IFunction
...
-        if (!isNativeNumber(arguments[1]))
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of second argument of function {}",
-                arguments[1]->getName(), getName());
-
-        if (number_of_arguments == 3 && !isNativeNumber(arguments[2]))
-            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of second argument of function {}",
-                arguments[2]->getName(), getName());
+        FunctionArgumentDescriptors mandatory_args{
+            {"s", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isStringOrFixedString), nullptr, "String"},
+            {"offset", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isNativeNumber), nullptr, "(U)Int8, (U)Int16, (U)Int32, (U)Int64 or Float"},
```
Comment (rschu1ze) [inline src/Functions/bitSlice.cpp:45]: `(U)Int8, (U)Int16, (U)Int32, (U)Int64 or Float` --> `(U)Int* or Float*`

Comment (Blokje5) [reply src/Functions/bitSlice.cpp:45]: Doesn't `(U)Int*` include 128 & 256 length integers? The function doesn't support those, hence the explicit references.

Comment (rschu1ze) [reply src/Functions/bitSlice.cpp:45]: Oh, you are right. To avoid repetition, maybe `(U)Int8/16/32/64 or Float`?

Comment (Blokje5) [reply src/Functions/bitSlice.cpp:45]: That makes sense. Do you want me to open a PR to update this description (this PR was already merged)?

Comment (alexkats) [reply src/Functions/bitSlice.cpp:45]: That's fine, I'll create a PR with some misc stuff including this one

**Pull request merged**